src/sp/it/pl/gui/pane/OverlayPane.kt

```diff
@@ -307,13 +307,23 @@ abstract class OverlayPane<in T>: StackPane() {
         }
     }
 
-    companion object {
+    companion object: MultiConfigurableBase("View") {
         private const val IS_SHOWN = "visible"
         private const val ROOT_STYLECLASS = "overlay-pane"
         private const val CONTENT_STYLECLASS = "overlay-pane-content"
+
+        @IsConfig(name = "Display method", group = "View", info = "Area of content. Screen provides more space than window, but can get in the way of other apps.")
```

Comment:
Because this separates data from metadata. Metadata are secondary and mostly for 'external' context (persisting, ui). Metadata are also always optional.

Alternative is a fluent builder pattern for configurable properties, but for the stated separation of concerns, I chose to avoid this solution. Creating configurable properties would quickly become a mess.